reporter/reporter.go

```diff
@@ -0,0 +1,237 @@
+package reporter
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	metrics "github.com/rcrowley/go-metrics"
+	"github.com/wavefronthq/wavefront-sdk-go/application"
+	"github.com/wavefronthq/wavefront-sdk-go/histogram"
+	wf "github.com/wavefronthq/wavefront-sdk-go/senders"
+)
+
+// WavefrontMetricsReporter report go-metrics to wavefront
+type WavefrontMetricsReporter interface {
+	Start()
+	Stop()
```

Comment:
Let's also add `Report()`. Allows users an option to not invoke start/stop but still report data.